### PR TITLE
#136900 - finalize timeline changes to use hapi and user id to create…

### PR DESCRIPTION
**Pull request by jesseeproductions**
… and get types, but Oauth for creating timeline events and fixes to RSVP

-Adds HAPI Key and User ID for Timeline Event Type Creation and Getting
-Changes Timeline Event to use the HAPI Key and User ID when setting up
-Fixes to RSVP
-Fixes to Timeline Event Template Extra Data
-Add time() to the timeline event custom id to make it unique and fix labeling of custom id
-Misc Text Changes
-Match custom date and datetime custom property names to the field type names

### CODE REVIEW

File: src/Tribe/Service_Provider.php [hunk 1]
```diff
@@ -63,7 +63,6 @@ public function register() {
 	 */
 	protected function hook() {
 
-		//todo change these to only load when required during Sprint 4
```
Comment (sc0ttkclark) [inline src/Tribe/Service_Provider.php]: Is this addressed?

File: src/Tribe/Properties/Event_Data.php [hunk 2]
```diff
@@ -191,6 +191,22 @@ public function get_ticket_values( $ticket_id, $attendee_id, $commerce, $name, $
 
 	}
 
+	/**
+	 * Determine if a Post Type is a Tribe Event.
```
Comment (sc0ttkclark) [inline src/Tribe/Properties/Event_Data.php]:
```suggestion
	 * Determine whether a post type is a TEC event.
```

File: src/Tribe/Properties/Event_Data.php [hunk 3]
```diff
@@ -191,6 +191,22 @@ public function get_ticket_values( $ticket_id, $attendee_id, $commerce, $name, $
...
+	 *
+	 * @param string $post_type Post Type Name.
```
Comment (sc0ttkclark) [inline src/Tribe/Properties/Event_Data.php]:
```suggestion
	 * @param string $post_type The post type name.
```

File: src/Tribe/Properties/Event_Data.php [hunk 4]
```diff
@@ -191,6 +191,22 @@ public function get_ticket_values( $ticket_id, $attendee_id, $commerce, $name, $
...
+	 *
+	 * @return bool True or false the post type.
```
Comment (sc0ttkclark) [inline src/Tribe/Properties/Event_Data.php]:
```suggestion
	 * @return bool Whether a post type is a TEC event.
```

File: src/Tribe/Service_Provider.php (hunk 1 above)
Comment (jesseeproductions) [reply src/Tribe/Service_Provider.php]: I left only the ones that have hooks that should run:
https://github.com/mt-support/tribe-ext-hubspot/blob/feature/137548-add-dev-hapi-key/src/Tribe/Service_Provider.php#L64

Let me know if there is a better way to do this. Thanks

**Comment by jesseeproductions** [comment]
My last commit removed time() from the timeline event id as it would make it almost impossible to every get id correct to get events ( if we would have that in the future ). Having the same id means it will update the existing timeline event, which I believe works as it would prevent duplicates.

**Pull request merged**